torch/_functorch/_aot_autograd/schemas.py

```diff
@@ -444,6 +445,32 @@ def __post_init__(self):
         # which tensors to be saved for the bwd graph.  num_forward captures
         # this information.
         self.num_forward = self.num_forward_returns + self.num_outputs_rng_offset
+        self.cache_safe = False
+
+    def make_cache_safe(self):
+        """
+        There are various fields in ViewAndMutationMeta that aren't serializable. This function is called after all tracing
+        is completed to simplify certain fields in the metadata so that they can be safely cached.
+
+        Doing so may lose information (in the case of traced_tangents), but none of the information is needed at runtime.
+        """
+        # TODO: This function is only a best effort: there are other fields that may not be cache safe
+        # (i.e., there's no guarantee that tensor_flatten() returns a serializable result), or that
+        # SubclassCreationMeta is cache safe.
+        assert not self.cache_safe
+        traced_tangent_metadata = []
+
+        def extract_metadata(t):
+            if isinstance(t, torch.Tensor) and is_traceable_wrapper_subclass(t):
+                (inner_tensors, flatten_spec) = t.__tensor_flatten__()  # type: ignore[attr-defined]
+                return (inner_tensors, flatten_spec)
+            else:
+                return None
+
+        self.traced_tangents = [
```

Comment:
I'll leave it to you, but I think it's a bit confusing that `ViewAndMutationMeta.traced_tangents` has a different type at compile time vs. runtime. Thoughts on a new class to represent the runtime metadata, or otherwise new fields on `ViewAndMutationMeta`?

Another case where we store extra stuff at runtime that we don't really need is `SubclassCreationMeta.original_subclass` ([code](https://github.com/pytorch/pytorch/blob/main/torch/_functorch/_aot_autograd/schemas.py#L213))

Where at runtime, all we need is the type of the subclass so we can invoke its unflatten method at runtime (today we call `type(subclass).__tensor_unflatten__`).

We also re-purpose the same fn at compile time today and actually do use the `original_subclass`.

Reply:
Sure, I can make a new field in ViewAndMutationMeta. I think making an entire new class for this is a bit overkill, especially if we're going to have @aorenste's changes that make FakeTensors with symints serializeable